.github/workflows/php.yml

```diff
@@ -16,3 +16,89 @@ jobs:
 
       - run: composer install --prefer-dist
       - run: ./vendor/bin/php-cs-fixer fix --dry-run --diff
+
+  phpstan:
+    name: PHP Static Analysis
+    runs-on: ubuntu-20.04
+    steps:
+        -   name: Setup PHP
+            uses: shivammathur/setup-php@v2
+            with:
+                php-version: '7.4'
+                extensions: mbstring, intl, gd, xml, dom, json, fileinfo, curl, zip, iconv
+        -   uses: actions/checkout@v2
+            with:
+                fetch-depth: 0
+        -   name: Composer Install
+            run: composer install --ansi --prefer-dist --no-interaction --no-progress --quiet
+
+        -   name: Run phpstan
+            run: ./vendor/bin/phpstan analyse -c phpstan.neon.dist
+
+  phpunit:
+    name: PHPUnit Tests
+    runs-on: ubuntu-latest
+    strategy:
+        matrix:
+            php: ['7.1', '7.2', '7.3', '7.4']
+    steps:
+        -   name: Setup PHP
+            uses: shivammathur/setup-php@v2
+            with:
+                php-version: ${{ matrix.php }}
+                extensions: mbstring, intl, gd, xml, dom, json, fileinfo, curl, zip, iconv
+        -   uses: actions/checkout@v2
+            with:
+                fetch-depth: 0
+
+        -   name: PrestaShop Configuration
+            run: cp tests-legacy/parameters.yml.travis app/config/parameters.yml
+
+        -   name: Composer Install
+            run: composer install --ansi --prefer-dist --no-interaction --no-progress --quiet
+
+        -   name: Run phpunit
+            run: ./vendor/bin/phpunit -c tests/Unit/phpunit.xml
+            env:
+                SYMFONY_DEPRECATIONS_HELPER: disabled
+
+  phpunit_with_db:
+      name: PHPUnit Tests (With DB)
+      runs-on: ubuntu-latest
+      strategy:
+          matrix:
+              php: ['7.1', '7.2', '7.3', '7.4']
+      steps:
+        -   name: Setup PHP
+            uses: shivammathur/setup-php@v2
+            with:
+                php-version: ${{ matrix.php }}
+                extensions: mbstring, intl, gd, xml, dom, json, fileinfo, curl, zip, iconv
+        -   name: Setup MySQL
+            uses: mirromutth/mysql-action@v1.1
+            with:
+                mysql version: '5.7'
+                mysql database: 'prestashop'
+                mysql root password: ''
+                mysql user: 'user'
+                mysql password: 'pass'
+
+        -   uses: actions/checkout@v2
+            with:
+                fetch-depth: 0
+
+        -   name: PrestaShop Configuration
+            run: cp tests-legacy/parameters.yml.travis app/config/parameters.yml
+
+        -   name: Composer Install
+            run: composer install --ansi --prefer-dist --no-interaction --no-progress --quiet
```

Comment:
shall we cache it? https://github.com/actions/cache/blob/main/examples.md#php---composer

Reply:
Good idea. 1m21 => 30s